src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java

```diff
@@ -0,0 +1,232 @@
+        XPayAuthResponse respone = new XPayAuthResponse();
+        respone.setRequestId(requestId);
+        if(StringUtils.isEmpty(errorMessage)) {
+            String urlRedirect = StringUtils.join(PGS_RESPONSE_URL_REDIRECT, requestId);
+            respone.setUrlRedirect(urlRedirect);
+        } else {
+            respone.setError(errorMessage);
+            log.info("END POST - " + REQUEST_PAYMENTS_XPAY);
+        }
+        return ResponseEntity.status(status).body(respone);
+    }
+
+    private ResponseEntity<XPayAuthResponse> createAuthPaymentXPay(XPayAuthRequest pgsRequest, String clientId, String mdcFields) {
+       log.info("START - XPay Request Payment Authorization ");
+        AuthPaymentXPayRequest xPayRequest = createXPayRequest(pgsRequest);
+
+        PaymentRequestEntity paymentRequestEntity = new PaymentRequestEntity();
+        String transactionId = pgsRequest.getIdTransaction();
+        try{
+            String authRequestJson = OBJECT_MAPPER.writeValueAsString(xPayRequest);
+            generateRequestEntity(clientId, mdcFields, transactionId, paymentRequestEntity, xPayRequest.getTimeStamp());
+            paymentRequestEntity.setJsonRequest(authRequestJson);
+            xPayRequest.setUrlRisposta(String.format(XPAY_RESPONSE_URL, paymentRequestEntity.getGuid()));
+        } catch (JsonProcessingException e) {
+            log.error(SERIALIZATION_ERROR_MSG, e);
+            return  createxPayAuthResponse(GENERIC_ERROR_MSG + transactionId, HttpStatus.INTERNAL_SERVER_ERROR, null);
+        }
+
+        try {
+            executeXPayAuthorizationCall(xPayRequest, paymentRequestEntity, transactionId);
+        } catch (Exception e) {
+            log.error(GENERIC_ERROR_MSG, e);
+            return createxPayAuthResponse(GENERIC_ERROR_MSG + transactionId, HttpStatus.INTERNAL_SERVER_ERROR, null);
+        }
+        log.info("END - XPay Request Payment Authorization ");
+        return createxPayAuthResponse(null, HttpStatus.OK, paymentRequestEntity.getGuid());
+    }
+
+    @Async
+    private void executeXPayAuthorizationCall(AuthPaymentXPayRequest xPayRequest, PaymentRequestEntity paymentRequestEntity, String transactionId) {
+        log.info("START - execute XPay payment authorization call for transactionId: " + transactionId);
+        AuthPaymentXPayResponse response;
+        PaymentResponseEntity paymentResponseEntity = new PaymentResponseEntity();
+        try {
+            response = service.postForObject(xPayRequest);
+        } catch (ResourceAccessException e) {
+            log.warn("Timeout exception for xpay api request call");
+            throw e;
+        } catch (HttpServerErrorException hse){
+            log.warn("HttpServerErrorException exception for xpay api request call");
+            throw hse;
+        }
+
+        paymentRequestEntity.setXpayHtml(response.getHtml());
```

Comment:
@DavideAmatoNexi better checking if the response object from XPay is not null.
Also, it is better to move the code under the catch clauses into the same try above